reana_commons/api_client.py

```diff
@@ -163,7 +163,7 @@ def submit(
         if htcondor_accounting_group:
             job_spec["htcondor_accounting_group"] = htcondor_accounting_group
 
-        if kubernetes_job_timeout:
+        if kubernetes_job_timeout is not None:
```

Comment:
Could you please explain the reasoning behind this change? Is this needed to cover e.g. `kubernetes_job_timeout = 0`?

Reply:
Yes.

Before this PR if you put `kubernetes_job_timeout=0`, it would not reach r-job-controller (because zero is treated as `False`). So, the r-job-controller will assume that no job timeout is supplied, so it will apply the default value specified in the helm chart. 

With this PR, `kubernetes_job_timeout=0` will be sent to r-job-controller. r-job-controller will validate the parameter and throw an error that value is less or equal to zero.

Reply:
I haven't actually tried what will happen if you enter something like this:

```yaml
kubernetes_job_timeout:
  something: 0
```

It should not pass, but curious what error will be.

Reply:
Coming back to the question. This is an edge case, and we do not expect people to pass zero value. But I just wanted to align things. If a user is using `kubernetes_job_timeout` and enters zero, we shouldn't silently use a default value, we should tell the user their value is incorrect.

Unfortunately, I missed it in my first PR. 

*Side note:* I like, for example, how Rust handles it. You cannot use `integer` in boolean expressions. You need to say `value == 0`, otherwise the compiler throws an error. Helps to avoid such bugs.

Reply:
> This is an edge case

Yes, it is indeed an edge case. But I can picture that some users might think that passing a zero value would disable the runtime check altogether. So it's good to catch these explicitly.

(I guess that elsewhere such as for `kubernetes_memory_limit` we are mostly good, since the values are usually strings such as "2G". Might be good to think about others anyway.)

> I like, for example, how Rust handles it

Go as well, non-booleans cannot be used in if conditions :smile: An inherent advantage of strongly-typed languages, I guess.

Reply:
> curious what error will be

```console
$ reana-client logs -w limit0
==> Workflow engine logs
Workflow exited unexpectedly.
Job submission error: {'kubernetes_job_timeout': ['kubernetes_job_timeout must be greater than 0.']}
```

We could perhaps format the error messages more nicely, but it'll do even like this. It's understandable.

Reply:
[This issue](https://github.com/reanahub/reana-workflow-controller/issues/426) should fix it. Try removing `REANA_RUNTIME_KUBERNETES_KEEP_ALIVE_JOBS_WITH_STATUSES: failed`, it will show workflow logs.